### PR TITLE
Fall back to CPU for deletion vectors writes on Databricks [databricks]

### PR DESCRIPTION
**Pull request by andygrove**
Related to https://github.com/NVIDIA/spark-rapids/issues/8654

This PR makes Databricks support consistent with our current OSS Delta Lake support by falling back to CPU for Delta Lake deletes against tables with deletion vectors enabled.

### CODE REVIEW

**Comment by andygrove** [comment]
build

File: integration_tests/src/main/python/delta_lake_delete_test.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ def write_func(spark, path):
 @ignore_order
 @pytest.mark.parametrize("use_cdf", [True, False], ids=idfn)
 @pytest.mark.skipif(is_before_spark_340(), reason="Deletion vectors new in Delta Lake 2.4 / Apache Spark 3.4")
-@pytest.mark.xfail(is_databricks122_or_later(), reason="https://github.com/NVIDIA/spark-rapids/issues/8654")
+@pytest.mark.skipif(is_databricks_runtime(), reason="Separate test exists for Databricks")
```
Comment (jlowe) [inline integration_tests/src/main/python/delta_lake_delete_test.py:91]: Odd to see a separate test for databricks.  Straightforward to create an `delta_supports_delete_vectors` boolean that can be used as a skip trigger across platforms.

File: integration_tests/src/main/python/delta_lake_delete_test.py [hunk 2]
```diff
@@ -99,14 +99,33 @@ def write_func(spark, path):
...
+                          use_cdf=use_cdf, enable_deletion_vectors=True)
+    def write_func(spark, path):
+        delete_sql="DELETE FROM delta.`{}`".format(path)
+        spark.sql(delete_sql)
+    with_cpu_session(setup_tables)
+    disable_conf = copy_and_update(delta_delete_enabled_conf,
+        {"spark.databricks.delta.delete.deletionVectors.persistent": "true"})
+
+    assert_gpu_fallback_write(write_func, read_delta_path, data_path,
+                              "LocalTableScanExec", disable_conf)
```
Comment (jlowe) [inline integration_tests/src/main/python/delta_lake_delete_test.py:127]: This doesn't seem to be the right thing to search for in a fallback.  LocalTableScanExec isn't what actually writes the deletion vector or performs the delete -- this should be looking for the CPU exec node that performs the delete, not one performing a read/scan.  There are other delete fallback tests that are passing on Databricks 12.2, curious why this wasn't modeled after those (or just reusing the non-Databricks fallback test which I would expect to work).

Comment (andygrove) [reply integration_tests/src/main/python/delta_lake_delete_test.py]: So this wasn't falling back at all, which led to my confusion. I needed to implement the fallback logic for DeleteCommandEdge in addition to DeleteCommand. This is now updated.

**Review by jlowe — APPROVED** [review]

**Comment by andygrove** [comment]
build

**Pull request merged**